app/services/enqueue_index_jobs.rb

```diff
@@ -0,0 +1,36 @@
+# Indexes a book version thru the SQS queuing system
+#
+# @params indexing_version e.g., Search::BookVersions::I1::IndexingStrategy::VERSION
+class EnqueueIndexJobs
```

Comment:
My reading of this class was that it contained all the work for the rake script, so the rake script could just call off to this PORO.  But looking at how it is initialized with a single indexing strategy I'm getting the feeling that may not be the case?

If this is intended to be just for one indexing strategy, that means that when running for multiple indexing strategies we'll be repulling and iterating through all the releases multiple times.  I'm not sure if that is the best.  

Above is discussed how the validation approach to checking if a BookIndexing already exists might not be ideal.  Also with our desire to delete no-longer-requested book/indexing strategy combos, an all-at-once approach like the following might be better (the below is very rough):

```ruby
class EnqueueIndexJobs
  def do_it
    rex_releases = OpenStax::RexReleases.new
    book_ids = rex_releases.map{ |release| release.books }.flatten

    book_indexings = BookIndexing.all # guessing that this is how you get all entries from Dynamo

    book_ids.each do |book_id|
      active_indexing_strategies.each do |indexing_strategy|
        # to avoid iterating through the entire book_indexings array over and over, probably good
        # to make a hash of book IDs to hashes of indexing strategies to BookIndexings

        book_indexing = that_crazy_hash[book_id].try(:[], indexing_strategy)

        if book_indexing.nil?
          # hey, we haven't done this one yet! 
          job = IndexingJob.new(book_version_id: ..., indexing_strategy: ...)
          @todo_jobs_queue.write(job) 
          BookIndexing.create!(...)
        else
          # we have done this one already, somehow note that it is still in demand
          # ones we don't mark don't have a book ID / indexing strategy combination
          # indicated in the released content, and should be deleted (below)
        end
    end

    # find all of our local book indexings that we didn't mark in demand and delete 
    # all of their indexes

  end
end
```